packages/core/src/state-management/state-manager.ts

```diff
@@ -78,16 +78,27 @@ export class StateManager {
    * @param state$
    * @param timeoutMillis
    */
-  async sync(state$: RunningState, timeoutMillis: number): Promise<void> {
+  async sync(state$: RunningState, timeoutMillis: number, hint?: CID): Promise<void> {
     const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id)
-    // We do not expect this promise to return anything, so set `defaultValue` to `undefined`
-    await lastValueFrom(
-      tip$.pipe(
-        takeUntil(timer(timeoutMillis)),
-        concatMap((tip) => this._handleTip(state$, tip))
-      ),
-      { defaultValue: undefined }
-    )
+    if (hint) {
```

Comment:
This makes `sync` more complicated and adds logic outside of its core function of syncing the tip from pubsub.  IMO I think this would be easier to follow if the body of `sync` was left alone and if the call site in `repository.load` explicitly called `handleTip` right after calling `sync` in the SYNC_ALWAYS case

Reply:
I would like to counter that proposal by noting, that it would make the loading process obviously longer.

Reply:
Sorry, how does this suggestion make the loading process longer?  I had assumed it was because the existing code in the PR applied the two commits in parallel, but now you're saying it happens serially, so I'm no longer clear why it's faster than this suggestion

Reply:
I'm confused, below you said `concatMap` processes things serially (which is important because if it didn't there'd be concurrent modifications to the state object), but here you're saying it can apply the hint commit in parallel to loading the tip from the network.  Aren't those things mutually exclusive?

Reply:
Oh nevermind, I think I get it now.  We kick off the query to the network first, then we go to the `concatMap` of applying the tips one at a time to the state object.

Reply:
Every individual commit is processed serially. You could do that while still waiting for the network to reply.

Reply:
![IMG_6905](https://user-images.githubusercontent.com/193527/188713309-5b9687c0-4d4c-4601-a851-16fd459f0607.jpg)

Reply:
Huh, GitHub should autorefresh the page on a comment.